### PR TITLE
ci: Add GH workflow to validate PR titles follow Conventional Commits.

### PR DESCRIPTION
**Pull request by Henry8192**
# Description
[Convention Commits](https://www.conventionalcommits.org) is a specification for writing commit messages (or in our case, PR titles) that makes it easy to see at a glance what change the commit makes which in turn makes it easier to generate release notes.

This PR adds a workflow to check PR titles match the spec.

# Validation performed
success: https://github.com/Henry8192/yscope-log-viewer/actions/runs/11846367297
failed: https://github.com/Henry8192/yscope-log-viewer/actions/runs/11846367174

## Summary by CodeRabbit

- **New Features**
	- Introduced a new GitHub Actions workflow to enforce conventional commit message standards for pull requests.
	- Updated the pull request template to include guidance on writing meaningful commit messages.

- **Documentation**
	- Revised the pull request template by removing the "References" section and adding instructions for the pull request title format.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
-# References
-<!-- Any issues or pull requests relevant to this pull request -->
+<!--
+Set the PR title to a meaningful commit message that:
+- follows the Conventional Commits specification (https://www.conventionalcommits.org).
+- is in imperative form.
 
+Template:
+<type>[optional scope]: <description>
+fix: Don't add implicit wildcards ('*') at the beginning and the end of a query (fixes #390).
```
Comment (junhaoliao) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: I think it's fine to let users figure out the exact format from https://www.conventionalcommits.org listed above, or we should probably explain what an "optional scope" is here as well.
```suggestion
Example:
fix: Don't add implicit wildcards ('*') at the beginning and the end of a query (fixes #390).
```

File: .github/workflows/pr-title-checks.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    permissions:
+      # For amannn/action-semantic-pull-request
+      pull-requests: "read"
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "amannn/action-semantic-pull-request@v5"
+        env:
+          GITHUB_TOKEN: "${{secrets.GITHUB_TOKEN}}"
+
+permissions: {}
```
Comment (junhaoliao) [inline .github/workflows/pr-title-checks.yaml:25]: Let put this before `concurrency` since this is at a meta level.

**Comment by junhaoliao** [comment]
For the PR title, how about:
```
ci: Add GH workflow to validate PR titles follow Conventional Commits.
```

File: .github/workflows/pr-title-checks.yaml (hunk 2 above)
Comment (Henry8192) [inline .github/workflows/pr-title-checks.yaml:25]:
```suggestion
permissions: {}

concurrency:
  group: "${{github.workflow}}-${{github.ref}}"

  # Cancel in-progress jobs for efficiency
  cancel-in-progress: true

jobs:
  conventional-commits:
    permissions:
      # For amannn/action-semantic-pull-request
      pull-requests: "read"
    runs-on: "ubuntu-latest"
    steps:
      - uses: "amannn/action-semantic-pull-request@v5"
        env:
          GITHUB_TOKEN: "${{secrets.GITHUB_TOKEN}}"

```

File: .github/workflows/pr-title-checks.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: "pr-title-checks"
+
+on:
+  pull_request_target:
+    types: ["edited", "opened", "reopened"]
+    branches: ["main"]
```
[the remark this answers is not in the document]
Comment (junhaoliao) [reply .github/workflows/pr-title-checks.yaml:6]: `permissions: {}` has been specified to address this.

**Review by junhaoliao — APPROVED** [review]

**Pull request merged**